apps/ledger-live-desktop/src/renderer/modals/SignMessage/steps/StepSummary.tsx

```diff
@@ -101,38 +96,19 @@ MessageProperties.displayName = "MessageProperties";
 export default function StepSummary({ account, message: messageData }: StepProps) {
   const [showAdvanced, setShowAdvanced] = useState(false);
   const mainAccount = getMainAccount(account, null);
-  const {
-    message,
-    fields,
-  }: {
-    message: string,
-    fields?: ReturnType<typeof getNanoDisplayedInfosFor712>,
-  } = useMemo(() => {
-    try {
-      if (mainAccount.currency.family === "ethereum") {
-        const parsedMessage: Record<string, unknown> =
-          typeof messageData.message === "string"
-            ? JSON.parse(messageData.message)
-            : messageData.message;
 
-        if (!isEIP712Message(parsedMessage)) {
-          throw new Error(); // Fallback to signing as string
-        }
-        return {
-          fields: getNanoDisplayedInfosFor712(parsedMessage),
-          message: JSON.stringify(messageData.message),
-        };
-      }
-      throw new Error(); // Fallback to signing as string
-    } catch (e) {
-      return {
-        message:
-          typeof messageData.message === "string"
-            ? messageData.message
-            : JSON.stringify(messageData.message),
-      };
-    }
-  }, [mainAccount.currency.family, messageData.message]);
+  const [messageProperties, setMessageProperties] = useState<{
+    message?: string;
+    fields?: NanoDisplayedInfoFor712;
+  }>({});
+
+  useEffect(() => {
+    getMessageProperties(mainAccount.currency, messageData).then(setMessageProperties);
+  }, [mainAccount.currency, messageData, setMessageProperties]);
+
+  const { message, fields } = messageProperties;
+
+  console.log({ fields });
```

Comment:
Drop it?

Reply:
Sorry, missed that one 👌

Reply:
Looks like I never pushed my rebase... 🤦‍♂️